src/pyHepGrid/src/runArcjob.py

```diff
@@ -76,12 +76,17 @@ def _run_XRSL(self, filename, test = False):
         # Speeds up submission (according to Stephen)
         if arc_direct and ".dur.scotgrid.ac.uk" in ce:
             cmd += " -S org.nordugrid.gridftpjob --direct "
-        output = util.getOutputCall(cmd.split())
-        jobid = output.split("jobid:")[-1].rstrip().strip()
-        return jobid
+        if include_retcode:
+            output = util.getOutputCall(cmd.split(), include_return_code=True)
+            jobid = output[0].split("jobid:")[-1].rstrip().strip()
+            return jobid, output[1]
+        else:
+            output = util.getOutputCall(cmd.split())
+            jobid = output.split("jobid:")[-1].rstrip().strip()
+            return jobid
```

Comment:
Are there functions which depend on receiving only one argument? Because I don't see the harm of always having the return code (and doing nothing with it when it is not needed)

Reply:
Yes, this was a hackaround that I pushed through to get everything working quickly when I implemented it. A cleanup everywhere _run_XRSL() is called to pass through jobid and output would be advisable, possibly even a good first issue for people new to the codebase?

Reply:
Yeah, it could be a good for starting a "low hanging fruit" tag :pager: